evalml/objectives/utils.py

```diff
@@ -214,3 +229,158 @@ def get_core_objectives(problem_type):
+    if exclude is not None:
+        for objective in exclude:
+            ex_obj = get_objective(objective)
+            if not ex_obj.is_defined_for_problem_type(problem_type):
+                raise ValueError(
+                    f"Objective to exclude {ex_obj} is not defined for {problem_type}",
+                )
+            if ex_obj.name not in default_objectives:
+                raise ValueError(
+                    f"Cannot exclude objective {ex_obj} because it is not in the default objectives",
+                )
+            exclude_objectives.append(ex_obj.name)
+
+    default_objectives.update(set(include_objectives))
+    return default_objectives - set(exclude_objectives)
+
+
+def normalize_objectives(objectives_to_normalize, max_objectives, min_objectives):
+    """Converts objectives from a [0, inf) scale to [0, 1] given a max and min for each objective.
+
+    Args:
+        objectives_to_normalize (dict[str,float]): A dictionary mapping objectives to values
+        max_objectives (dict[str,float]): The mapping of objectives to the maximum values for normalization
+        min_objectives (dict[str,float]): The mapping of objectives to the minimum values for normalization
+
+    Returns:
+        A dictionary mapping objective names to their new normalized values
+    """
+    normalized = {}
+    for objective_name, val in objectives_to_normalize.items():
+        objective_obj = get_objective(objective_name)
+        # Only normalize objectives that are not bounded like percentages
+        # R2 also does not get normalized as it's essentially bounded like a percentage,
+        # and we want to penalize aggressively when R2 is negative
+        if objective_obj.is_bounded_like_percentage or objective_obj.name == "R2":
+            normalized[objective_name] = val
+            continue
+        max_val, min_val = (
+            max_objectives[objective_name],
+            min_objectives[objective_name],
+        )
+        if max_val == min_val:
+            normal = 1
+        else:
+            normal = (val - min_val) / (max_val - min_val)
+            if not objective_obj.greater_is_better:
+                normal = 1 - normal
+        normalized[objective_name] = normal
+    return normalized
+
+
+def recommendation_score(
+    objectives,
+    prioritized_objective=None,
+    prioritized_weight=0.5,
+):
+    """Computes a recommendation score for a model given scores for a group of objectives.
+
+    Args:
+        objectives (dict[str,float]): A dictionary mapping objectives to their values
+        prioritized_objective (str): An optional name of a priority objective that should be given heavier weight
+            than the other objectives contributing to the score. Defaults to None, where all objectives are
+            weighted equally.
+        prioritized_weight (float): The weight to attribute to the prioritized objective, if it exists.
+            Defaults to 0.5.
```

Comment:
You might want to specify here that it should be between 0 and 1 and that, probably unnecessary, higher is weightier?